run_nvfuser_tests.py

```diff
@@ -0,0 +1,591 @@
+
+    except KeyboardInterrupt:
+        # Kill any running processes
+        for process in current_processes.values():
+            if process is not None:
+                process.kill()
+        raise
+    finally:
+        # Ensure all processes are cleaned up
+        for process in current_processes.values():
+            if process is not None:
+                process.kill()
+
+
+def collect_test_failures(log_dir, dry_run=False):
+    """Scan log files for test failures and collect context"""
+    if dry_run:
+        print(
+            "\nWould scan log files for failures and create failure_summary.txt with:"
+        )
+        print("- 5 lines of context before each failure")
+        print("- 20 lines of context after each failure")
+        print("- Coverage for both GTest and Pytest failures")
+        print("- Timeout failure information")
+        return False
+
+    failure_summary = []
+
+    for log_file in log_dir.glob("*.log"):
+        with open(log_file, "r") as f:
+            lines = f.readlines()
+
+        for i, line in enumerate(lines):
+            # Look for GTest failures between square brackets
+            if "[" in line and "]" in line:
+                result_part = line[line.find("[") : line.find("]") + 1]
+                if "FAILED" in result_part or "TIMEOUT" in result_part:
+                    failure_summary.append(
+                        f"\n=== GTest Failure in {log_file.name} ==="
+                    )
+                    # Collect 5 lines before and 20 lines after the failure
+                    start = max(0, i - 5)
+                    end = min(len(lines), i + 21)
+                    failure_summary.extend(lines[start:end])
+                continue  # Skip other checks for this line if it was a bracketed result
+
+            # Look for pytest failures
+            if "FAILED" in line and "test" in line.lower():
+                failure_summary.append(f"\n=== Pytest Failure in {log_file.name} ===")
+                # Collect 5 lines before and 20 lines after the failure
+                start = max(0, i - 5)
+                end = min(len(lines), i + 21)
+                failure_summary.extend(lines[start:end])
+
+            # Look for timeout failures
+            if "ERROR: Test timed out" in line:
+                failure_summary.append(f"\n=== Timeout in {log_file.name} ===")
+                failure_summary.append(line)
+
+    # Write failure summary if there were any failures
+    if failure_summary:
+        with open(log_dir / "failure_summary.txt", "w") as f:
+            f.write("=== Test Failure Summary ===\n")
+            f.write("".join(failure_summary))
+
+        return True
+    return False
+
+
+def get_available_gpus():
+    """Check how many NVIDIA GPUs are available"""
+    try:
+        # Run nvidia-smi to get GPU count
+        result = subprocess.run(
+            ["nvidia-smi", "--query-gpu=gpu_name", "--format=csv,noheader"],
```

Comment:
Consider `subprocess.check_output("nvidia-smi -L | wc -l", shell=True)` which is simpler.